src/entities/logical/randomEventGenerator.gd

```diff
@@ -5,6 +5,7 @@ signal random_event(event)
 var rng: RandomNumberGenerator = RandomNumberGenerator.new()
 var bias: int = 0 # "Bad Luck Protection" to increase chance of generating event with time
 var accumulatedDelta: float = 0
+var isEventHappening: bool = false
```

Comment:
where is this checked? Would it make sense to have `if (!isEventHappening)` in the beginning of `_process`?

Reply:
Yup, adding this, that's the idea behind the on random event completed signal in `world.gd`